pkg/errors/consts.go

```diff
@@ -0,0 +1,51 @@
+package errors
```

Comment:
My comments regarding your errors library:

1. I do appreciate having a class for each kind error as opposed to just using `errors.New` - it enables users to classify and test errors by their type.
2. However - There's currently no difference between "ErrorCode" & "ErrorMessage". I think it's obvious that "Conflict" = "The resource already exists" & "NotFound" = "The resource not found". To me the "ErrorMessage" list is redundant. Instead it would be nice if each error type would enable passing in a detailed message on initiation (for example `err = NewPermitForbiddenError("this is not allowed to access that")`).
3. The `ErrorType` is a cool idea - but you don't really use it (only used type is `GENERAL_ERROR` so you probably don't need it (principle of YAGNI). Even if necessary - those kind of classifications are more traditionally achieved by inheritance.

Reply:
In general, we decided to follow (in a minimal way) after Stripe's error handling - https://stripe.com/docs/api/errors
I might not using the ErrorType here, but our users might use it ?
I want to give them the ability to remediate their issues better and help them differentiate between the various errors.
And I agree, I should use the different kind of ErrorTypes.

I'll add the custom message for errors.

Reply:
Cool. But even though - I think having the "ErrorMessage" which basically is the same as "ErrorCode" (bullet 2 in my initial comment) doesn't look serious. (or did you fix that as well?)

Reply:
Didn't fix this.
It's part of the general schema that we decided to go with, i'll go over it with Asaf.